js/src/ankerSnapshot.ts

```diff
@@ -0,0 +1,206 @@
+import {
+  AccountInfo,
+  Connection,
+  ParsedAccountData,
+  PublicKey,
+} from '@solana/web3.js';
+import BN from 'bn.js';
+import { deserializeUnchecked } from 'borsh';
+import { AnkerSnapshot, getSolido, ProgramAddresses, StLamports } from '.';
+
+/**
+ * Anker Program State
+ *
+ * Reference:
+ * https://github.com/ChorusOne/solido/blob/73040002ddbb62a3cee93107d03871f848ecd1e0/program/src/state.rs#L187
+ */
+export class Anker {
+  solido_program_id: PublicKey;
+  solido: PublicKey;
+  b_sol_mint: PublicKey;
+  token_swap_pool: PublicKey;
+  terra_rewards_destination: any;
+  wormhole_parameters: WormholeParameters;
+  metrics: Metrics;
+  self_bump_seed: number;
+  mint_authority_bump_seed: number;
+  reserve_authority_bump_seed: number;
+  st_sol_reserve_account_bump_seed: number;
+  ust_reserve_account_bump_seed: number;
+
+  constructor(data: any) {
+    try {
+      const parsedData: Anker = {
+        ...data,
+        solido_program_id: new PublicKey(data.solido_program_id.toArray('le')),
+        solido: new PublicKey(data.solido.toArray('le')),
+        b_sol_mint: new PublicKey(data.b_sol_mint.toArray('le')),
+        token_swap_pool: new PublicKey(data.token_swap_pool.toArray('le')),
+      };
+
+      Object.assign(this, parsedData);
+    } catch (error) {
+      console.log('error', error);
+    }
+  }
+}
+
+class WormholeParameters {
+  core_bridge_program_id: PublicKey;
+  token_bridge_program_id: PublicKey;
+
+  constructor(data: any) {
+    try {
+      const parsedData = {
+        ...data,
+        core_bridge_program_id: new PublicKey(
+          data.core_bridge_program_id.toArray('le')
+        ),
+        token_bridge_program_id: new PublicKey(
+          data.token_bridge_program_id.toArray('le')
+        ),
+      };
+
+      Object.assign(this, parsedData);
+    } catch (error) {
+      console.log('error', error);
+    }
+  }
+}
+
+class Metrics {
+  swapped_rewards_st_sol_total: BN;
```

Comment:
These numbers are in lamports and µUST, I would rename them to `swapped_rewards_st_lamports_total` and `swapped_rewards_micro_ust_total`, if we can’t use strong types here.